detector/PID/ARC_geo_v0.cpp

```diff
@@ -0,0 +1,871 @@
+//----------------------------------
+//         ARC detector v0
+//----------------------------------
+
+/*!
+ *  \brief     Detector constructor of barrel of ARC detector
+ *  \details   This code creates full geometry of ARC detector
+ *             Evolved from the pfRICH example in DD4hep.
+ *  \author    Alvaro Tolosa-Delgado alvaro.tolosa.delgado@cern.ch
+ *  \author    Martin Tat            martin.tat@cern.ch
+ *  \version   0
+ *  \date      2023
+ *  \pre       DD4hep compiled with Geant4+Qt
+ *  \bug       Walls do not reflect optical photons. Hard-coded values in many places.
+ */
+
+#include "DD4hep/DetFactoryHelper.h"
+#include "DD4hep/OpticalSurfaces.h"
+#include "DD4hep/Printout.h"
+
+using namespace dd4hep;
+
+// #define DUMP_SENSOR_POSITIONS
+
+/// Function to build ARC endcaps
+static Ref_t create_arc_endcap_cell(Detector &desc, xml::Handle_t handle, SensitiveDetector sens)
```

Comment:
It is worth mentioning that there are two detector elements declared in this builder (e.g. in the README). I guess you have common code between barrel and endcap and that you prefer to have everything in one single file? If not, you could just have two cpp files, one for the barrel, one for the endcap.